bot.go

```diff
@@ -47,6 +49,41 @@ func (b Bot) Listen(subscription chan<- Message, timeout time.Duration) {
 	}()
 }
 
+// ListenWebHook starts HTTPS server and register webHooks in Telegram for PUSH
+// updates. certFile and keyFile - filenames containing a certificate and matching private key for the server
+// See https://golang.org/pkg/net/http/#ListenAndServeTLS
+func (b Bot) ListenWebHook(subscription chan<- Message, publicAddress, path, certFile, keyFile string) error {
```

Comment:
Yes and no:
1. Future users must always remember, that SSL connection is required - using just http.Server is not requires this fact
2. Probably we may extend this feature in next PR if it really needs